### ISSUE TITLE
Gør det muligt kun at vise aktiviteter for en enkelt forening eller afdeling [Feature]

### ISSUE THREAD

**Issue by JakobLibak** [CodingPirates/forenings_medlemmer#663]
# Formål
Det skal være muligt for en afdeling eller forening at vise sine aktiviteter separat, så en kaptajn kan sende et direkte link eller man kan trykke på afdelingens navn for at se mere.

Det skal eksempelvis være muligt at lave en url til `/activity/department/[id]` eller `/activity/union/|id]` og så vil den vise en liste over alle nuværende aktiviteter for pågældende afdeling eller forening.

Under oversigten over afdelinger skal hver enkelt afdeling have en knap med [Se denne afdelings aktiviteter]

# Tekniske ændringer
- `urls.py`skal opdateres til at håndtere department og union som extension under /activity
- ny underside hvor en afdelings eller forenings aktiviteter vises.
- Lav nye undersider `/department` og `/union`, hvor der vises detaljer om forening/afdeling og alle igangværende aktiviteter vises.
- Vis hvilke personer fra familien der står på venteliste til afdeling.
- Aktiviteter der ikke er åbne for tilmelding skal beskrives som "Denne aktivitet er ikke åben for tilmelding. Du kan skrive dig op til næste gang vi opretter et nyt hold" Derefter knap for hver person i familien der kan opskrives på venteliste. Knap skal ikke vises hvis person allerede står på venteliste.

**Comment by jarl-dk** [comment]
@JakobLibak : Jeg går ud fra at når du har sat denne til "In progress" og tildelt den til mig, så er det fordi det er næste skridt.

**Comment by JakobLibak** [comment]
@jarl-dk Ja, du må gerne se på denne. Giver beskrivelsen mening?

**Comment by jarl-dk** [comment]
@JakobLibak 
Ja, sådan nogenlunde... Men jeg synes det lyder som om der er flere features beskrevet.

# Feature 1a (en side for hver forening)
Som almindelig besøgende:
* ny underside hvor en \[\] forenings aktiviteter vises.
* Lav nye undersider `/union`, hvor der vises detaljer om forening og alle igangværende aktiviteter vises.
* Det skal eksempelvis være muligt at lave en url \[cut\] og så vil den vise en liste over alle nuværende aktiviteter for pågældende  forening.

# Feature 1b, fortsættelse af 1: (Siden for hver forening)
Som bruger af systemet:
* Vis hvilke personer fra familien der står på venteliste til foreningen.
* Aktiviteter der ikke er åbne for tilmelding skal beskrives som "Denne aktivitet er ikke åben for tilmelding. Du kan skrive dig op til næste gang vi opretter et nyt hold" Derefter knap for hver person i familien der kan opskrives på venteliste. Knap skal ikke vises hvis person allerede står på venteliste.


# Feature 2a (en side for hver afdeling)
Som almindelig besøgende:
* Under oversigten over afdelinger skal hver enkelt afdeling have en knap med [Se denne afdelings aktiviteter]
* ny underside hvor en afdelings aktiviteter vises.
* Lav nye undersider `/department`, hvor der vises detaljer om afdeling og alle igangværende aktiviteter vises.
* Det skal eksempelvis være muligt at lave en url \[cut\] og så vil den vise en liste over alle nuværende aktiviteter for pågældende afdeling.

# Feature 2b, fortsættelse af 1: (Siden for hver afdeling)
Som bruger af systemet:
* Vis hvilke personer fra familien der står på venteliste til afdeling.
* Aktiviteter der ikke er åbne for tilmelding skal beskrives som "Denne aktivitet er ikke åben for tilmelding. Du kan skrive dig op til næste gang vi opretter et nyt hold" Derefter knap for hver person i familien der kan opskrives på venteliste. Knap skal ikke vises hvis person allerede står på venteliste.


# Feature 3 (en side for hver aktivitet)
Som almindelig besøgende:
* Det skal være muligt for en afdeling eller forening at vise sine aktiviteter separat, så en kaptajn kan sende et direkte link eller man kan trykke på afdelingens navn for at se mere.

Det sidste i feature 3 "eller man kan trykke på afdelingens navn for at se mere" forstår jeg ikke helt.

Hvilken en har højeste prioritet?

**Comment by JakobLibak** [comment]
Jeg foreslår at vi ændrer scope for denne feature, så det kun bliver til en ny afdelingsside i første omgang:

# Overordnet
Hver afdeling skal have sin egen side under `/department/[afdelingsnavn]` (brug ikke id, men afdelingsnavn strippet for danske karakterer).

# Indhold på side
## Afdelingsnavn
Tekst:
Her kan du se information om [name].
Kontaktperson: [responsible_name] + [department_email]
Adresse: [address]
Åbningstid: [open_hours]
Afdelingens hjemmeside vises som link, hvis der er registreret et weblink [website]
Beskrivelse af afdeling: [description]
## Tilmeldte aktiviteter
Genbrug listen fra `/activities`, men vis kun aktiviteter for denne afdeling
## Invitationer
Genbrug listen fra `/activities`, men filtrer for denne afdeling.
## Aktiviteter
Genbrug listen fra `/activities`, men vis kun aktiviteter for den aktuelle afdeling og vis derudover _alle_ aktive aktiviteter. For lukkede aktiviteter skriv at _Tilmelding kræver invitation._ et passende sted i listen.
# Øvrige ændringer
De eksisterende lister med afdelinger har pt. et hyperlink til afdelingens hjemmeside. Det skal ændres, så der i stedet linkes til denne nye side. Dette gælder for `/activities` (nyt link), `/department_signup` (ret eksisterende link i begge lister)

@jarl-dk

**Comment by jarl-dk** [comment]
Er der nogen grund til at undgå danske tegn i stien af URL'en?

**Comment by JakobLibak** [comment]
Nej, det er bare mig, der tænker at det ikke vil virke. Men hvis det virker, så kører vi bare videre med danske tegn. @jarl-dk

**Comment by jarl-dk** [comment]
OK. Det går jeg i gang med...

**Comment by jarl-dk** [comment]
@JakobLibak : Jeg går ud fra at vi tager denne med
> Under oversigten over afdelinger skal hver enkelt afdeling have en knap med [Se denne afdelings aktiviteter]

Ellers er det svært at komme hen til siden :-)

**Comment by JakobLibak** [comment]
Ja tak. :-) @jarl-dk

**Comment by jarl-dk** [comment]
@JakobLibak
Kan du bekræfte:
at "aktive aktiviteter." betyder aktiviteter med `end_date` i fremtiden?
at "lukkede aktiviteter" betyder `open_invite` er `False`?

**Comment by jarl-dk** [comment]
@JakobLibak : Du skriver
> Genbrug listen fra `/activities`, men vis kun aktiviteter for den aktuelle afdeling og vis derudover alle aktive aktiviteter. For lukkede aktiviteter skriv at Tilmelding kræver invitation. et passende sted i listen.

Men Activities har 3 lister. Jeg tænker at alle 3 lister skal vises på afdeling-siden og bgrænses til afdelingen. Er det listen "Åbne aktiviteter" der skal omdøbes til "Alle aktiviteter" og også omfavne lukkede aktiviteter og aktiviteter hvor tilmeldingsfriten er overskredet? Eller hvordan?

**Comment by JakobLibak** [comment]
@jarl-dk 
>at "aktive aktiviteter." betyder aktiviteter med end_date i fremtiden?

Ja
>at "lukkede aktiviteter" betyder open_invite er False?

Ja

>Men Activities har 3 lister. Jeg tænker at alle 3 lister skal vises på afdeling-siden og bgrænses til afdelingen. Er det listen "Åbne aktiviteter" der skal omdøbes til "Alle aktiviteter" og også omfavne lukkede aktiviteter og aktiviteter hvor tilmeldingsfriten er overskredet? Eller hvordan?

Ja

Hvad du skriver giver mening for mig, så jeg tænker at vi virker afstemte. Men skriv hvis jeg mangler at uddybe noget.

**Comment by jarl-dk** [comment]
Skal alle historiske aktiviteter (`end_date` i fortiden) også være med i "Alle aktiviteter", nu hvor `signup_closing` i fortiden er med?

**Comment by JakobLibak** [comment]
> Skal alle historiske aktiviteter (`end_date` i fortiden) også være med i "Alle aktiviteter", nu hvor `signup_closing` i fortiden er med?

Nej. Den regel skal fortsætte. Aktiviteten skal være synlig indtil den er slut.

@jarl-dk

**Comment by rasmusselsmark** [comment]
Det er blevet nemmere med filter-funktionalitet på https://members.codingpirates.dk/activities/. Besluttet at give prioritet C
